nominal/core.py

```diff
@@ -250,7 +237,7 @@ def update(
         update_dataclass(self, dataset, fields=self.__dataclass_fields__)
         return self
 
-    def add_csv_to_dataset(self, path: Path | str, timestamp_column: str, timestamp_type: TimestampColumnType) -> None:
+    def add_csv_to_dataset(self, path: Path | str, timestamp_column: str, timestamp_type: TypedTimeDomain) -> None:
```

Comment:
do we still want the Domain naming here / will this be exposed to users? would lean towards something api-like (e.g TimestampType) for user-facing functions for consistency

Reply:
sure, not opposed to that!

Reply:
well, I am opposed to it in the sense that I think "type" is the wrong word to denote these as it's not the datatype but the format...but we're already there so idk

Reply:
I do want to expand these signatures to include the string types.